2023-04-19 17:54:26,720 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-19 17:54:26,720 INFO [33mPress CTRL+C to quit[0m
2023-04-20 16:27:17,053 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-20 16:27:17,053 INFO [33mPress CTRL+C to quit[0m
2023-04-20 16:56:57,106 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-20 16:56:57,107 INFO [33mPress CTRL+C to quit[0m
2023-04-21 03:08:40,022 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 03:08:40,023 INFO [33mPress CTRL+C to quit[0m
2023-04-21 05:02:13,056 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 05:02:13,056 INFO [33mPress CTRL+C to quit[0m
2023-04-21 05:04:31,019 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 05:04:31,020 INFO [33mPress CTRL+C to quit[0m
2023-04-21 05:08:33,148 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 05:08:33,148 INFO [33mPress CTRL+C to quit[0m
2023-04-21 05:12:09,706 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 05:12:09,706 INFO [33mPress CTRL+C to quit[0m
2023-04-21 05:18:48,122 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 05:18:48,123 INFO [33mPress CTRL+C to quit[0m
2023-04-21 05:20:04,089 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 05:20:04,090 INFO [33mPress CTRL+C to quit[0m
2023-04-21 05:33:41,074 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 05:33:41,074 INFO [33mPress CTRL+C to quit[0m
2023-04-21 05:41:13,041 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 05:41:13,041 INFO [33mPress CTRL+C to quit[0m
2023-04-21 05:41:41,021 INFO 127.0.0.1 - - [21/Apr/2023 05:41:41] "POST /api/v1/save_info?saved_existing_data=TRUE HTTP/1.1" 200 -
2023-04-21 05:43:05,756 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 05:43:05,756 INFO [33mPress CTRL+C to quit[0m
2023-04-21 05:43:17,236 INFO 127.0.0.1 - - [21/Apr/2023 05:43:17] "POST /api/v1/save_info?saved_existing_data=TRUE HTTP/1.1" 200 -
2023-04-21 06:03:10,373 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 06:03:10,373 INFO [33mPress CTRL+C to quit[0m
2023-04-21 06:04:02,128 INFO SaveSalaryInformationServiceImpl: save salary information
2023-04-21 06:04:02,141 INFO SaveSalaryInformationServiceImpl: Successfully saved the information's provided
2023-04-21 06:04:02,143 INFO 127.0.0.1 - - [21/Apr/2023 06:04:02] "POST /api/v1/save_info?saved_existing_data=FALSE HTTP/1.1" 200 -
2023-04-21 06:42:37,019 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 06:42:37,020 INFO [33mPress CTRL+C to quit[0m
2023-04-21 07:00:46,027 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 07:00:46,027 INFO [33mPress CTRL+C to quit[0m
2023-04-21 07:02:07,092 INFO 127.0.0.1 - - [21/Apr/2023 07:02:07] "POST /api/v1/encoded?encode_specific=True HTTP/1.1" 200 -
2023-04-21 07:02:14,873 INFO 127.0.0.1 - - [21/Apr/2023 07:02:14] "POST /api/v1/encoded?encode_specific=FALSE HTTP/1.1" 200 -
2023-04-21 07:03:52,058 INFO 127.0.0.1 - - [21/Apr/2023 07:03:52] "POST /api/v1/encoded?encode_specific=FALSE HTTP/1.1" 200 -
2023-04-21 07:04:48,508 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 07:04:48,508 INFO [33mPress CTRL+C to quit[0m
2023-04-21 07:04:59,705 INFO 127.0.0.1 - - [21/Apr/2023 07:04:59] "POST /api/v1/encoded?encode_specific=FALSE HTTP/1.1" 200 -
2023-04-21 07:05:32,276 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 07:05:32,276 INFO [33mPress CTRL+C to quit[0m
2023-04-21 07:06:09,922 INFO 127.0.0.1 - - [21/Apr/2023 07:06:09] "POST /api/v1/encoded?encode_specific=FALSE HTTP/1.1" 200 -
2023-04-21 07:08:14,736 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 07:08:14,736 INFO [33mPress CTRL+C to quit[0m
2023-04-21 07:08:23,932 INFO 127.0.0.1 - - [21/Apr/2023 07:08:23] "POST /api/v1/encoded?encode_specific=FALSE HTTP/1.1" 200 -
2023-04-21 07:08:47,516 INFO 127.0.0.1 - - [21/Apr/2023 07:08:47] "POST /api/v1/encoded?encode_specific=TRUE HTTP/1.1" 200 -
2023-04-21 07:10:07,270 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 07:10:07,271 INFO [33mPress CTRL+C to quit[0m
2023-04-21 07:10:13,040 INFO 127.0.0.1 - - [21/Apr/2023 07:10:13] "POST /api/v1/encoded?encode_specific=YES HTTP/1.1" 200 -
2023-04-21 07:10:17,609 INFO 127.0.0.1 - - [21/Apr/2023 07:10:17] "POST /api/v1/encoded?encode_specific=NO HTTP/1.1" 200 -
2023-04-21 07:13:59,988 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 07:13:59,988 INFO [33mPress CTRL+C to quit[0m
2023-04-21 07:14:06,001 INFO 127.0.0.1 - - [21/Apr/2023 07:14:06] "POST /api/v1/encoded?encode_specific=NO HTTP/1.1" 200 -
2023-04-21 07:20:43,213 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 07:20:43,213 INFO [33mPress CTRL+C to quit[0m
2023-04-21 07:23:53,094 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 07:23:53,094 INFO [33mPress CTRL+C to quit[0m
2023-04-21 07:24:03,446 INFO 127.0.0.1 - - [21/Apr/2023 07:24:03] "POST /api/v1/encoded?encode_specific=NO HTTP/1.1" 200 -
2023-04-21 07:28:15,805 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 07:28:15,805 INFO [33mPress CTRL+C to quit[0m
2023-04-21 07:28:26,090 INFO 127.0.0.1 - - [21/Apr/2023 07:28:26] "POST /api/v1/encoded?encode_specific=NO HTTP/1.1" 200 -
2023-04-21 07:29:14,970 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 07:29:14,970 INFO [33mPress CTRL+C to quit[0m
2023-04-21 07:29:19,112 INFO 127.0.0.1 - - [21/Apr/2023 07:29:19] "POST /api/v1/encoded?encode_specific=NO HTTP/1.1" 200 -
2023-04-21 16:04:02,947 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:04:02,947 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:04:25,554 INFO 127.0.0.1 - - [21/Apr/2023 16:04:25] "POST /api/v1/encoded?encode_specific=NO HTTP/1.1" 200 -
2023-04-21 16:07:31,447 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:07:31,447 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:07:35,724 INFO 127.0.0.1 - - [21/Apr/2023 16:07:35] "POST /api/v1/encoded?encode_specific=NO HTTP/1.1" 200 -
2023-04-21 16:07:54,098 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(db_session,
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 66, in save
    if len(encoded_service_impl) != 0:
TypeError: object of type 'Response' has no len()
2023-04-21 16:07:54,112 INFO 127.0.0.1 - - [21/Apr/2023 16:07:54] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-21 16:09:21,942 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:09:21,942 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:12:31,923 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:12:31,923 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:12:48,588 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(db_session,recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 66, in save
    if len(encoded_service_impl) != 0:
TypeError: object of type 'Response' has no len()
2023-04-21 16:12:48,590 INFO 127.0.0.1 - - [21/Apr/2023 16:12:48] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-21 16:15:03,704 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:15:03,704 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:20:43,933 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:20:43,933 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:21:44,886 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:21:44,886 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:22:23,209 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:22:23,209 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:22:26,482 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(db_session,recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 66, in save
    if len(encoded_service_impl) != 0:
TypeError: object of type 'Response' has no len()
2023-04-21 16:22:26,483 INFO 127.0.0.1 - - [21/Apr/2023 16:22:26] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-21 16:22:35,416 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:22:35,416 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:23:16,135 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:23:16,136 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:23:35,699 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(db_session,recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 66, in save
    if len(encoded_service_impl) != 0:
TypeError: object of type 'EncodedSalaryConversion' has no len()
2023-04-21 16:23:35,714 INFO 127.0.0.1 - - [21/Apr/2023 16:23:35] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-21 16:24:37,567 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:24:37,567 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:27:25,916 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:27:25,916 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:29:55,415 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:29:55,416 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:31:38,027 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:31:38,027 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:31:45,151 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(db_session,recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 66, in save
    print(encoded_service_impl[0]['encoded_value'])
TypeError: list indices must be integers or slices, not str
2023-04-21 16:31:45,153 INFO 127.0.0.1 - - [21/Apr/2023 16:31:45] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-21 16:31:53,085 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:31:53,085 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:32:06,821 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(db_session,recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 66, in save
    print(encoded_service_impl[0]['encoded_value'])
TypeError: list indices must be integers or slices, not str
2023-04-21 16:32:06,836 INFO 127.0.0.1 - - [21/Apr/2023 16:32:06] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-21 16:32:33,922 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(db_session,recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 66, in save
    print(encoded_service_impl[0]['encoded_value'])
TypeError: list indices must be integers or slices, not str
2023-04-21 16:32:33,929 INFO 127.0.0.1 - - [21/Apr/2023 16:32:33] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-21 16:35:11,754 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:35:11,754 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:37:19,966 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:37:19,966 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:37:26,166 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 89, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(db_session,recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 106, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'
2023-04-21 16:37:26,167 INFO 127.0.0.1 - - [21/Apr/2023 16:37:26] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-21 16:39:23,657 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:39:23,657 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:39:43,063 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:39:43,063 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:41:52,813 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:41:52,813 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:42:01,863 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 90, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(db_session,recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 107, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'
2023-04-21 16:42:01,867 INFO 127.0.0.1 - - [21/Apr/2023 16:42:01] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-21 16:43:46,989 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:43:46,989 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:43:54,105 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 90, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(db_session,recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 107, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'
2023-04-21 16:43:54,107 INFO 127.0.0.1 - - [21/Apr/2023 16:43:54] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-21 16:44:47,485 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:44:47,485 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:49:48,732 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:49:48,733 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:52:28,617 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:52:28,617 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:52:41,114 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:52:41,114 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:53:35,618 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:53:35,619 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:55:38,971 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:55:38,971 INFO [33mPress CTRL+C to quit[0m
2023-04-21 16:57:36,482 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 16:57:36,482 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:00:20,844 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:00:20,844 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:18:32,733 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:18:32,733 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:21:56,316 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:21:56,316 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:23:04,940 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 90, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(db_session,recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 107, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'
2023-04-21 17:23:04,943 INFO 127.0.0.1 - - [21/Apr/2023 17:23:04] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-21 17:24:55,689 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:24:55,689 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:26:32,549 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:26:32,549 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:31:05,030 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:31:05,030 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:31:17,560 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 90, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(db_session,recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 107, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'
2023-04-21 17:31:17,562 INFO 127.0.0.1 - - [21/Apr/2023 17:31:17] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-21 17:32:32,788 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:32:32,789 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:34:37,712 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:34:37,713 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:38:58,292 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:38:58,292 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:39:05,445 INFO 127.0.0.1 - - [21/Apr/2023 17:39:05] "POST /api/v1/get_recommendation HTTP/1.1" 200 -
2023-04-21 17:42:08,338 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:42:08,338 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:46:54,831 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:46:54,831 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:48:02,058 INFO 127.0.0.1 - - [21/Apr/2023 17:48:02] "POST /api/v1/get_recommendation HTTP/1.1" 200 -
2023-04-21 17:52:15,948 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:52:15,949 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:52:41,065 INFO 127.0.0.1 - - [21/Apr/2023 17:52:41] "POST /api/v1/get_recommendation HTTP/1.1" 200 -
2023-04-21 17:56:36,471 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:56:36,471 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:56:42,679 INFO 127.0.0.1 - - [21/Apr/2023 17:56:42] "PUT /api/v1/rating/1?rating=4 HTTP/1.1" 200 -
2023-04-21 17:58:33,366 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:58:33,366 INFO [33mPress CTRL+C to quit[0m
2023-04-21 17:58:40,297 INFO 127.0.0.1 - - [21/Apr/2023 17:58:40] "PUT /api/v1/rating/1?rating=4 HTTP/1.1" 200 -
2023-04-21 17:59:54,057 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 17:59:54,057 INFO [33mPress CTRL+C to quit[0m
2023-04-21 18:00:06,017 INFO 127.0.0.1 - - [21/Apr/2023 18:00:06] "POST /api/v1/save_info?saved_existing_data=TRUE HTTP/1.1" 200 -
2023-04-21 18:00:24,729 INFO SaveSalaryInformationServiceImpl: save salary information
2023-04-21 18:00:24,732 INFO SaveSalaryInformationServiceImpl: Successfully saved the information's provided
2023-04-21 18:00:24,733 INFO 127.0.0.1 - - [21/Apr/2023 18:00:24] "POST /api/v1/save_info?saved_existing_data=FALSE HTTP/1.1" 200 -
2023-04-21 18:01:07,037 INFO 127.0.0.1 - - [21/Apr/2023 18:01:07] "POST /api/v1/encoded?encode_specific=YES HTTP/1.1" 200 -
2023-04-21 18:01:21,052 INFO 127.0.0.1 - - [21/Apr/2023 18:01:21] "POST /api/v1/encoded?encode_specific=NO HTTP/1.1" 200 -
2023-04-21 18:01:34,287 INFO 127.0.0.1 - - [21/Apr/2023 18:01:34] "POST /api/v1/get_recommendation HTTP/1.1" 200 -
2023-04-21 21:37:39,120 INFO This is an informational message
2023-04-21 21:37:47,006 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-21 21:37:47,006 INFO [33mPress CTRL+C to quit[0m
2023-04-23 05:01:09,157 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 05:01:09,158 INFO [33mPress CTRL+C to quit[0m
2023-04-23 05:01:39,445 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 05:01:39,446 INFO [33mPress CTRL+C to quit[0m
2023-04-23 05:02:50,068 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 05:02:50,068 INFO [33mPress CTRL+C to quit[0m
2023-04-23 05:03:10,925 INFO 127.0.0.1 - - [23/Apr/2023 05:03:10] "POST /api/v1/get_recommendation HTTP/1.1" 200 -
2023-04-23 05:09:39,190 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 05:09:39,190 INFO [33mPress CTRL+C to quit[0m
2023-04-23 05:09:46,413 INFO 127.0.0.1 - - [23/Apr/2023 05:09:46] "POST /api/v1/get_recommendation HTTP/1.1" 200 -
2023-04-23 05:10:37,896 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 05:10:37,896 INFO [33mPress CTRL+C to quit[0m
2023-04-23 05:46:44,772 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 05:46:44,773 INFO [33mPress CTRL+C to quit[0m
2023-04-23 15:02:35,675 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 15:02:35,675 INFO [33mPress CTRL+C to quit[0m
2023-04-23 15:04:05,883 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 15:04:05,883 INFO [33mPress CTRL+C to quit[0m
2023-04-23 15:04:52,135 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 15:04:52,135 INFO [33mPress CTRL+C to quit[0m
2023-04-23 15:12:31,970 INFO 127.0.0.1 - - [23/Apr/2023 15:12:31] "POST /api/v1/encoded?encode_specific=YES HTTP/1.1" 200 -
2023-04-23 15:12:37,322 INFO 127.0.0.1 - - [23/Apr/2023 15:12:37] "POST /api/v1/encoded?encode_specific=NO HTTP/1.1" 200 -
2023-04-23 15:12:45,926 INFO 127.0.0.1 - - [23/Apr/2023 15:12:45] "POST /api/v1/save_info?saved_existing_data=TRUE HTTP/1.1" 200 -
2023-04-23 15:13:12,212 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 91, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 75, in get_recommendation
    created_response = get_recommendation_result.save(db_session, recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 110, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'
2023-04-23 15:13:12,231 INFO 127.0.0.1 - - [23/Apr/2023 15:13:12] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-23 15:15:17,839 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 15:15:17,839 INFO [33mPress CTRL+C to quit[0m
2023-04-23 15:15:27,976 INFO 127.0.0.1 - - [23/Apr/2023 15:15:27] "POST /api/v1/get_recommendation HTTP/1.1" 200 -
2023-04-23 15:26:38,711 INFO 127.0.0.1 - - [23/Apr/2023 15:26:38] "PUT /api/v1/rating/460?rating=4 HTTP/1.1" 200 -
2023-04-23 15:32:28,038 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 15:32:28,039 INFO [33mPress CTRL+C to quit[0m
2023-04-23 15:32:35,510 INFO 127.0.0.1 - - [23/Apr/2023 15:32:35] "POST /api/v1/get_recommendation HTTP/1.1" 200 -
2023-04-23 17:25:41,262 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 17:25:41,262 INFO [33mPress CTRL+C to quit[0m
2023-04-23 17:27:22,305 INFO 127.0.0.1 - - [23/Apr/2023 17:27:22] "GET /api/v1/create_table/recommendation?information_provider=YES HTTP/1.1" 200 -
2023-04-23 17:32:37,201 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 17:32:37,201 INFO [33mPress CTRL+C to quit[0m
2023-04-23 17:32:45,976 INFO 127.0.0.1 - - [23/Apr/2023 17:32:45] "GET /api/v1/create_table/recommendation?information_provider=YES HTTP/1.1" 200 -
2023-04-23 17:51:41,219 INFO 127.0.0.1 - - [23/Apr/2023 17:51:41] "GET /api/v1/create_table/recommendation?information_provider=YES&ENCODE_DATA=TRUE HTTP/1.1" 200 -
2023-04-23 19:13:10,863 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 19:13:10,863 INFO [33mPress CTRL+C to quit[0m
2023-04-23 19:13:37,582 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 70, in save
    if len(encoded_service_impl) != 0:
TypeError: object of type 'Response' has no len()
2023-04-23 19:13:37,599 INFO 127.0.0.1 - - [23/Apr/2023 19:13:37] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-23 19:14:13,020 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 19:14:13,020 INFO [33mPress CTRL+C to quit[0m
2023-04-23 19:15:02,291 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 70, in save
    if len(encoded_service_impl) != 0:
TypeError: object of type 'Response' has no len()
2023-04-23 19:15:02,297 INFO 127.0.0.1 - - [23/Apr/2023 19:15:02] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-23 19:19:26,217 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 70, in save
    if len(encoded_service_impl) != 0:
TypeError: object of type 'Response' has no len()
2023-04-23 19:19:26,220 INFO 127.0.0.1 - - [23/Apr/2023 19:19:26] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-23 19:19:40,958 INFO 127.0.0.1 - - [23/Apr/2023 19:19:40] "GET /api/v1/create_table/recommendation?information_provider=YES&ENCODE_DATA=TRUE HTTP/1.1" 200 -
2023-04-23 19:20:35,436 INFO 127.0.0.1 - - [23/Apr/2023 19:20:35] "GET /api/v1/create_table/recommendation?information_provider=YES&ENCODE_DATA=FALSE HTTP/1.1" 200 -
2023-04-23 19:21:20,202 INFO 127.0.0.1 - - [23/Apr/2023 19:21:20] "GET /api/v1/create_table/recommendation?information_provider=YES HTTP/1.1" 200 -
2023-04-23 19:21:36,186 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 19:21:36,187 INFO [33mPress CTRL+C to quit[0m
2023-04-23 19:23:01,541 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 19:23:01,541 INFO [33mPress CTRL+C to quit[0m
2023-04-23 19:23:38,485 INFO 127.0.0.1 - - [23/Apr/2023 19:23:38] "GET /api/v1/create_table/recommendation?information_provider=YES HTTP/1.1" 200 -
2023-04-23 19:23:55,663 INFO 127.0.0.1 - - [23/Apr/2023 19:23:55] "GET /api/v1/create_table/recommendation?information_provider=NO HTTP/1.1" 200 -
2023-04-23 19:41:16,200 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 19:41:16,201 INFO [33mPress CTRL+C to quit[0m
2023-04-23 19:41:28,407 INFO 127.0.0.1 - - [23/Apr/2023 19:41:28] "GET /api/v1/create_table/recommendation?information_provider=NO HTTP/1.1" 200 -
2023-04-23 19:41:38,368 INFO 127.0.0.1 - - [23/Apr/2023 19:41:38] "GET /api/v1/create_table/recommendation?information_provider=YES HTTP/1.1" 200 -
2023-04-23 19:42:02,187 ERROR Exception on /api/v1/create_table/recommendation [GET]
Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1826, in full_dispatch_request
    return self.finalize_request(rv)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1845, in finalize_request
    response = self.make_response(rv)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2137, in make_response
    raise TypeError(
TypeError: The view function for 'create_table' did not return a valid response. The function either returned None or ended without a return statement.
2023-04-23 19:42:02,195 INFO 127.0.0.1 - - [23/Apr/2023 19:42:02] "[35m[1mGET /api/v1/create_table/recommendation?information_provider=YESENCODE_DATA=TRUE HTTP/1.1[0m" 500 -
2023-04-23 19:42:13,113 INFO 127.0.0.1 - - [23/Apr/2023 19:42:13] "GET /api/v1/create_table/recommendation?information_provider=YES&ENCODE_DATA=TRUE HTTP/1.1" 200 -
2023-04-23 19:42:22,714 INFO 127.0.0.1 - - [23/Apr/2023 19:42:22] "GET /api/v1/create_table/recommendation?information_provider=YES&ENCODE_DATA=FALSE HTTP/1.1" 200 -
2023-04-23 19:43:18,614 INFO 127.0.0.1 - - [23/Apr/2023 19:43:18] "GET /api/v1/create_table/recommendation?information_provider=ENCODE_DATA&encode_conversion=TRUE HTTP/1.1" 200 -
2023-04-23 19:43:24,811 INFO 127.0.0.1 - - [23/Apr/2023 19:43:24] "GET /api/v1/create_table/recommendation?information_provider=ENCODE_DATA&encode_conversion=FALSE HTTP/1.1" 200 -
2023-04-23 19:44:07,726 INFO 127.0.0.1 - - [23/Apr/2023 19:44:07] "GET /api/v1/create_table/recommendation?information_provider=ENCODE_DATA HTTP/1.1" 200 -
2023-04-23 19:45:02,740 INFO 127.0.0.1 - - [23/Apr/2023 19:45:02] "POST /api/v1/save_info?saved_existing_data=TRUE HTTP/1.1" 200 -
2023-04-23 19:45:14,265 INFO 127.0.0.1 - - [23/Apr/2023 19:45:14] "POST /api/v1/encoded?encode_specific=YES HTTP/1.1" 200 -
2023-04-23 19:45:19,115 INFO 127.0.0.1 - - [23/Apr/2023 19:45:19] "POST /api/v1/encoded?encode_specific=NO HTTP/1.1" 200 -
2023-04-23 19:46:51,464 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 92, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 109, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'
2023-04-23 19:46:51,470 INFO 127.0.0.1 - - [23/Apr/2023 19:46:51] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-23 19:49:03,259 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 19:49:03,259 INFO [33mPress CTRL+C to quit[0m
2023-04-23 19:49:20,753 ERROR Exception on /api/v1/get_recommendation [POST]
Traceback (most recent call last):
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 92, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\shaza\PycharmProjects\pythonProject\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\main.py", line 67, in get_recommendation
    created_response = get_recommendation_result.save(recommendation_system)
  File "D:\FYP\Implementation\FYP-Implementation-for-deloyment\salary-recommendation-backend\ServiceImplementation\GetRecommendationServiceImpl.py", line 109, in save
    recommendation['content_based_filtering'] = dataframe.to_dict(orient='records')
AttributeError: 'NoneType' object has no attribute 'to_dict'
2023-04-23 19:49:20,756 INFO 127.0.0.1 - - [23/Apr/2023 19:49:20] "[35m[1mPOST /api/v1/get_recommendation HTTP/1.1[0m" 500 -
2023-04-23 19:50:05,349 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 19:50:05,349 INFO [33mPress CTRL+C to quit[0m
2023-04-23 19:58:57,740 INFO [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
2023-04-23 19:58:57,741 INFO [33mPress CTRL+C to quit[0m
2023-04-23 19:59:59,457 INFO 127.0.0.1 - - [23/Apr/2023 19:59:59] "POST /api/v1/get_recommendation HTTP/1.1" 200 -
2023-04-23 20:00:36,822 INFO 127.0.0.1 - - [23/Apr/2023 20:00:36] "PUT /api/v1/rating/460?rating=4 HTTP/1.1" 200 -
2023-04-23 20:00:54,116 INFO SaveSalaryInformationServiceImpl: save salary information
2023-04-23 20:00:54,171 INFO SaveSalaryInformationServiceImpl: Successfully saved the information's provided
2023-04-23 20:00:54,172 INFO 127.0.0.1 - - [23/Apr/2023 20:00:54] "POST /api/v1/save_info?saved_existing_data=FALSE HTTP/1.1" 200 -
